pkg/apiserver/handlers.go

```diff
@@ -188,11 +190,11 @@ func (t *timeoutHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	done := make(chan struct{}, 1)
+	done := make(chan struct{})
 	tw := newTimeoutWriter(w)
 	go func() {
 		t.handler.ServeHTTP(tw, r)
```

Comment:
Will this bring down the server? I thought server was recovered from panic. Also the log in the original issue showed that the API server recovered from the panic. We just did not handle the request correctly.